### PR TITLE
chore: limit unsound unwraps

### PR DESCRIPTION
**Pull request by LesnyRumcajs**
## Summary of changes

Changes introduced in this pull request:

- iterated over some unsound explicit unwraps/expects in the code and found those; there were genuine panic conditions although rather hard to exploit (you could mess up your host clock... to crash your own Forest :clown_face: ). The FVM2 address conversion is a more severe one that could surface in case of an archival node sitting on some pre NV18 state.

## Reference issue to close (if applicable)

Part of https://github.com/ChainSafe/forest/issues/7448

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

### Outside contributions

- [ ] This pull request is based on an issue that a maintainer has accepted (see [Before Opening a Pull Request][4]).
- [ ] I have read and agree to the [CONTRIBUTING][2] document.
- [ ] I have read and agree to the [AI Policy][3] document. I understand that failure to comply with the guidelines will lead to rejection of the pull request.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md
[2]: https://github.com/ChainSafe/forest/blob/main/CONTRIBUTING.md
[3]: https://github.com/ChainSafe/forest/blob/main/AI_POLICY.md
[4]: https://github.com/ChainSafe/forest/blob/main/CONTRIBUTING.md#before-opening-a-pull-request

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of invalid message serialization and address conversions, preventing unexpected crashes during execution and synchronization.
  - Added safer processing for unsupported addresses in legacy state trees.
  - Prevented timestamp-related failures when the system clock reports errors or exceeds supported limits.
  - Improved tolerance of invalid, negative, fractional, non-finite, or oversized percentage values.
  - Corrected storage-key handling for EVM state lookups.

- **Tests**
  - Added coverage for percentage conversion edge cases and round-trip behavior.

### CODE REVIEW

File: src/interpreter/vm.rs [hunk 1]
```diff
@@ -429,12 +429,17 @@ impl VM {
     pub fn apply_implicit_message(&mut self, msg: &Message) -> ApplyResult {
         let start = Instant::now();
 
-        // raw_length is not used for Implicit messages.
-        let raw_length = to_vec(msg).expect("encoding error").len();
+        // Every FVM version charges a zero `GasCharge` for `ApplyKind::Implicit` and never reads
+        // `raw_length`, so don't pay to serialize the message just to measure it.
+        let raw_length = 0;
```
Comment (LesnyRumcajs) [inline src/interpreter/vm.rs:434]: I've no clue why we decided to pay the price of serialization just to get a length that's discarded anyway. Even the comment there was acknowledging that. This should optimize every implicit message call which is nice (but not game-changing).

https://github.com/filecoin-project/ref-fvm/blob/4b552ce48f97ee52cad4224ef3bda2b26e7b4498/fvm/src/executor/default.rs#L243-L248
https://github.com/filecoin-project/ref-fvm/blob/142c2522e3bda59c9ac159a7403ee8c46a1458e1/fvm/src/executor/default.rs#L341-L346
https://github.com/filecoin-project/ref-fvm/blob/58e02e80cb0f933a2841137955175361de632261/fvm/src/executor/default.rs#L358-L363

File: src/lotus_json/percent.rs [hunk 2]
```diff
@@ -18,14 +17,13 @@ impl HasLotusJson for Percent {
...
-            "ratio may only have two decimals: {lotus_json}"
-        );
-        Percent(u64::try_from(scaled as i64).expect("ratio out of range"))
+        // Scaling via the decimal representation avoids `1.10 * 100.0 == 110.00000000000001`.
+        let scaled = format!("{lotus_json}e2").parse::<f64>().unwrap_or(f64::NAN);
+        // Lotus rejects these outright, but this conversion is infallible by trait contract.
+        if !scaled.is_finite() || scaled < 0.0 || scaled.trunc() != scaled {
+            tracing::warn!("ratio must be a non-negative multiple of 0.01, coercing: {lotus_json}");
+        }
+        Percent(scaled as u64) // saturates at both ends, NaN maps to 0
```
[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply src/lotus_json/percent.rs:26]: Verified against the current code — this doesn't reproduce, so skipping.

The premise is that `format!("{}", 1e300_f64)` yields `"1e300"`. It doesn't. `lotus_json` here is an `f64`, not the raw JSON text, and Rust's `Display` for floats never emits scientific notation (`{:e}` is the exponential format):

```
1e300    => Display "10000000000000000000…" (301 chars) ; +"e2" parses => 18446744073709551615
f64::MAX => Display "17976931348623157000…" (309 chars) ; +"e2" parses => 18446744073709551615
```

rust-lang/rust#24556 is a feature *request* for exponential `Display`, not a description of current behaviour.

Most directly: the `overflow_saturates` case added in this PR asserts exactly the outcome being asked for (`1e300 -> Percent(u64::MAX)`) and passes — 21/21 green, including `infinity` and `negative_infinity`.

Switching to numeric scaling would be a regression rather than a fix. `x * 100.0` is wrong for 573 of the 10,000 two-decimal ratios in `[0, 100)` — e.g. `0.29 * 100.0 == 28.999999999999996`, which truncates to 28 — and it also produces 1146 spurious "not two decimals" warnings. Avoiding that is precisely why the decimal-string form is used, and it mirrors Lotus (`chain/types/percent.go` does the same `+"e2"`).

Worth noting for the record: Lotus appends `e2` to the **raw JSON bytes**, so it genuinely does break on exponent-form input (`"1.5e10"` -> `"1.5e10e2"` -> parse error). Forest routes through `f64` `Display` first and is immune. That looks like the real bug this finding describes, just in the other implementation.

**Review by sudo-shashank — APPROVED** [review]
